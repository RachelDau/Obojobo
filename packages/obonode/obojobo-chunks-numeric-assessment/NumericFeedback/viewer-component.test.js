import React from 'react'
import renderer from 'react-test-renderer'
import NumericFeedback from './viewer-component'
import OboModel from 'obojobo-document-engine/src/scripts/common/models/obo-model'

// require('./viewer') // used to register this oboModel
// require('../MCChoice/viewer') // used to register the MCChoice chunk as a dep
// require('../NumericFeedback/viewer') // // dependency on Obojobo.Chunks.MCAssessment.NumericFeedback
// require('obojobo-chunks-text/viewer') // used to register the Text chunk as a dep

describe.skip('NumericFeedback', () => {
	test('NumericFeedback component for a correct item', () => {
		const moduleData = {
			focusState: {}
		}
		OboModel.create({
			id: 'parent',
			type: 'ObojoboDraft.Chunks.MCAssessment.MCChoice',
			content: {
				score: 100
			},
			children: [
				{
					id: 'feedback',
					type: 'ObojoboDraft.Chunks.MCAssessment.NumericFeedback',
					children: [
						{
							id: 'choice1-feedback-text',
							type: 'ObojoboDraft.Chunks.Text',
							content: {
								textGroup: [
									{
										text: {
											value: 'Example Text 2'
										}
									}
								]
							}
						}
					]
				}
			]
		})
		const component = renderer.create(
			<NumericFeedback moduleData={moduleData} model={OboModel.models.feedback} />
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericFeedback component for an incorrect item', () => {
		const moduleData = {
			focusState: {}
		}
		OboModel.create({
			id: 'parent',
			type: 'ObojoboDraft.Chunks.MCAssessment.MCChoice',
			content: {
				score: 0
			},
			children: [
				{
					id: 'feedback',
					type: 'ObojoboDraft.Chunks.MCAssessment.NumericFeedback',
					children: [
						{
							id: 'choice1-feedback-text',
							type: 'ObojoboDraft.Chunks.Text',
							content: {
								textGroup: [
									{
										text: {
											value: 'Example Text 2'
										}
									}
								]
							}
						}
					]
				}
			]
		})
		const component = renderer.create(
			<NumericFeedback moduleData={moduleData} model={OboModel.models.feedback} />
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
