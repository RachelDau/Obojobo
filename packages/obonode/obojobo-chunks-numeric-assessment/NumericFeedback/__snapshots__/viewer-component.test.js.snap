// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericFeedback NumericFeedback component for a correct item 1`] = `
<div
  className="component obojobo-draft--chunks--numeric-assessment--numeric-feedback"
  data-focussed={false}
  data-id="feedback"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.NumericAssessment.NumericFeedback"
  id="obo-feedback"
  tabIndex={-1}
>
  <span
    className="for-screen-reader-only"
  >
    . Feedback for this answer:
  </span>
</div>
`;

exports[`NumericFeedback NumericFeedback component for an incorrect item 1`] = `
<div
  className="component obojobo-draft--chunks--numeric-assessment--numeric-feedback"
  data-focussed={false}
  data-id="feedback"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.NumericAssessment.NumericFeedback"
  id="obo-feedback"
  tabIndex={-1}
>
  <span
    className="for-screen-reader-only"
  >
    . Feedback for this answer:
  </span>
</div>
`;
